infoschema/perfschema/tables.go

```diff
@@ -148,3 +162,100 @@ func (vt *perfSchemaTable) IterRecords(ctx sessionctx.Context, startKey kv.Key,
 	}
 	return nil
 }
+
+func dataForTiKVProfileCPU(ctx sessionctx.Context) ([][]types.Datum, error) {
+	servers, err := infoschema.GetTiKVServerInfo(ctx)
+	failpoint.Inject("mockTiKVNodeStatusAddress", func(val failpoint.Value) {
+		// The cluster topology is injected by `failpoint` expression and
+		// there is no extra checks for it. (let the test fail if the expression invalid)
+		if s := val.(string); len(s) > 0 {
+			servers = servers[:0]
+			for _, server := range strings.Split(s, ";") {
+				parts := strings.Split(server, ",")
+				servers = append(servers, infoschema.ServerInfo{
+					ServerType: parts[0],
+					Address:    parts[1],
+					StatusAddr: parts[2],
+				})
+			}
+			// erase error
+			err = nil
+		}
+	})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	type result struct {
+		addr string
+		rows [][]types.Datum
+		err  error
+	}
+
+	wg := sync.WaitGroup{}
+	ch := make(chan result, len(servers))
+	for _, server := range servers {
+		statusAddr := server.StatusAddr
+		if len(statusAddr) == 0 {
+			ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("TiKV node %s does not contain status address", statusAddr))
```

Comment:
should use `server.IP`here? cause `len(statusAddr) == 0` and output %s means nothing.

Reply:
Nice catch.